spec/java_buildpack/framework/open_telemtry_javaagent_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+require 'component_helper'
+require 'java_buildpack/framework/opentelemetry_javaagent'
+require 'java_buildpack/util/tokenized_version'
+
+describe JavaBuildpack::Framework::OpenTelemetryJavaagent do
+  include_context 'with component help'
+
+  let(:configuration) { { 'version' => '1.27.0' } }
+  let(:vcap_application) { { 'application_name' => 'GreatServiceTM' } }
+
+  it 'does not detect without otel-collector service bind' do
+    expect(component.detect).to be_nil
+  end
+
+  context 'when detected' do
+
+    before do
+      allow(services).to receive(:one_service?).with(/otel-collector/).and_return(true)
+    end
+
+    it 'detects with opentelemetry-javaagent' do
+      expect(component.detect).to eq("opentelemetry-javaagent=#{version}")
```

Comment:
The detection output also must match the file/class name (dashes replacing underscores) so this will need to be updated too.

```suggestion
      expect(component.detect).to eq("open-telemetry-javaagent=#{version}")
```